CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -348,8 +348,8 @@ public static function createPremiumTrxn($params) {
         'to_financial_account_id' => CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($params['financial_type_id'], $toFinancialAccountType),
         'from_financial_account_id' => CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($params['financial_type_id'], $fromFinancialAccountType),
         'trxn_date' => date('YmdHis'),
-        'total_amount' => !empty($params['cost']) ? $params['cost'] : 0,
-        'currency' => CRM_Utils_Array::value('currency', $params),
+        'total_amount' => $params['cost'] ?? 0,
+        'currency' => $params['currency'] ?? NULL,
```

Comment:
This one snuck in here. The !empty one isn't really equivalent. The original is probably slightly better.

I'm still going through these. I'll have an hour tomorrow where I'll just be "waiting" and should be able to finish then.

Reply:
Thanks for your work on this @demeritcowboy.
I snuck that one in manually (it was just changed from `CRM_Utils_Array::value` to `!empty()` in #16729) My thinking is that this param is a number, so it will either be a number or null. The coalesce operator therefore works fine, as the number 0 will stay 0, and null will be converted to 0.

Reply:
Ok so I had some extra time because hockey is cancelled indefinitely. Stay safe everybody!

I've looked at all of the commits and there's just this `!empty` one. Since I have all this extra time, I'll be nitpicky about it, and this whole exercise is somewhat academic anyway.

The question here is what is the expectation when $params['cost'], which sounds like something that should be a number, is not a number or null. For example if it's TRUE, then is it ok if that starts showing up as `$1.00` somewhere downstream? If it's the letter 'a', should it let that thru as letter 'a', silently change it to 0, or die? One possibility is something like `CRM_Utils_Type::validate($params['cost'] ?? 0, 'Money')`, which is an alias for CRM_Utils_Rule::numeric(), and throws an exception if it's not numeric.

I guess what I'm saying is I'd lean to either:

* Fail loudly when the data is clearly expected to be a certain thing,

OR

* Convert all possible edge cases to the same nullish/zeroish/blank value.

Right now, this variable seems definitely expected to be numeric, possibly null/missing, but the proposed change silently creates potentially odd things for different types of unexpected values, which is one (of a couple) thing(s) that has always bugged me about Array::value(), and what started this recent marathon. So if it's not going to be equivalent, then we might as well consider what it should really be.

Reply:
@demeritcowboy given  the  difficulty of merging such a huge commit  do you think we can merge this commit & then  do a follow up on that one?) - Or alternatively  perhaps we can merge a reviewers commit with all the approved changes in it?

Reply:
Yep I have no objection to merging the lot + followup (or not).

Reply:
OK - I'll do it. This PR definitely convinced me to abandon this  round of CRM_Utils_Array hammering but it will be good to have this one merged

Reply:
Yes @colemanw if you can hold off on more for a while, or sneak them one-by-one into other PRs as they come up.

Reply:
Lol - I try to only ever do a handful in one PR - ones like this are easy to make the PR because it's pattern matching. But hard to review & merge